lib/cucumber/formatter/junit.rb

```diff
@@ -157,6 +160,21 @@ def build_testcase(result, scenario_designation, output)
         @current_feature_data[:tests] += 1
       end
 
+      def get_testcase_attributes(classname, name, duration, filename)
+        attributes = { classname: classname, name: name, time: format('%<duration>.6f', duration: duration) }
```

Comment:
Minor optimisation

```
{ class...... }.tap do |attributes|
  attributes[:file] = filename if add_fileattribute?
end
```

Reply:
I like that method of using `.tap` much better. Thank you for the suggestion.